From: Sven Verdoolaege <skimo@kotnet.org>
Date: Sun, 12 Jan 2014 13:35:00 +0000 (+0100)
Subject: cloog_domain_cube: reimplement using documented functions
X-Git-Url: http://repo.or.cz/w/cloog.git/commitdiff_plain/b561f860f2fefa84459750d576807d214e4aad97?hp=afad88a8fd4e8c070c7a70e7a4e148af7232d06c

cloog_domain_cube: reimplement using documented functions

The original implementation used the undocumented
isl_basic_set_interval function, which will be removed
in the next release of isl.

Signed-off-by: Sven Verdoolaege <skimo@kotnet.org>
Signed-off-by: Cedric Bastoul <cedric.bastoul@unistra.fr>
---

diff --git a/source/isl/domain.c b/source/isl/domain.c
index d11da7b..620584d 100644
--- a/source/isl/domain.c
+++ b/source/isl/domain.c
@@ -1389,20 +1389,20 @@ CloogDomain *cloog_domain_cube(CloogState *state,
 				int dim, cloog_int_t min, cloog_int_t max)
 {
 	int i;
-	struct isl_basic_set *cube;
-	struct isl_basic_set *interval;
-	struct isl_basic_set_list *list;
+	isl_space *space;
+	isl_set *cube;
 
 	if (dim == 0)
 		return cloog_domain_universe(state, dim);
 
-	interval = isl_basic_set_interval(state->backend->ctx, min, max);
-	list = isl_basic_set_list_alloc(state->backend->ctx, dim);
-	for (i = 0; i < dim; ++i)
-		list = isl_basic_set_list_add(list, isl_basic_set_copy(interval));
-	isl_basic_set_free(interval);
-	cube = isl_basic_set_list_product(list);
-	return cloog_domain_from_isl_set(isl_set_from_basic_set(cube));
+	space = isl_space_set_alloc(state->backend->ctx, 0, dim);
+	cube = isl_set_universe(space);
+	for (i = 0; i < dim; ++i) {
+		cube = isl_set_lower_bound(cube, isl_dim_set, i, min);
+		cube = isl_set_upper_bound(cube, isl_dim_set, i, max);
+	}
+
+	return cloog_domain_from_isl_set(cube);
 }
 
 
